R/parents.R

```diff
@@ -0,0 +1,137 @@
+#' Getter and setter for specific parent
+#'
+#' @param join_keys_obj (`join_keys`) object to retrieve.
+#' @param dataset_name (`character(1)`)
+#'
+#' @export
+#'
+#' @examples
+#' jk <- join_keys()
+#' parent(jk, "ds1")
+#' parents(jk) <- list("ds2" = "ds3")
+#' parent(jk, "ds2")
+parent <- function(join_keys_obj, dataset_name) {
+  if (missing(dataset_name)) {
+    return(NULL)
+  }
+  checkmate::assert_string(dataset_name)
+  # assert join_keys_obj is performed by parents()
+  parents(join_keys_obj)[[dataset_name]]
+}
+
+#' Getter and setter functions for parents attribute of `join_keys`
+#'
+#' @param join_keys_obj (`join_keys`) object to retrieve or manipulate.
+#' @return a list of `character` representing the parents.
+#'
+#' @export
+parents <- function(join_keys_obj) {
```

Comment:
It may have been explained somewhere but why create a generic with only one method? Or is this missing a `.default` that will raise an error?

Reply:
It gives flexibility to add a `teal_data`, which it should have.

Added in 1d40996f